sdk/src/main/scala/com/ubirch/defaults/DefaultExternalGetter.scala

```diff
@@ -0,0 +1,49 @@
+package com.ubirch.defaults
+
+import java.util.Date
+
+import com.typesafe.config.Config
+import com.ubirch.api.{ ExternalResponseData, ExternalStateGetter, HMAC }
+import org.apache.http.HttpResponse
+import org.apache.http.client.methods.{ HttpPost, HttpUriRequest }
+import org.apache.http.impl.client.{ CloseableHttpClient, HttpClients }
+import org.apache.http.util.EntityUtils
+
+import javax.inject.{ Inject, Singleton }
+import org.apache.http.entity.ByteArrayEntity
+
+@Singleton
+class DefaultExternalGetter @Inject() (config: Config, HMAC: HMAC) extends ExternalStateGetter {
+
+  private val httpclient: CloseableHttpClient = HttpClients.createDefault
+
+  private final val TOKEN_MANAGER_ENDPOINT: String = config.getString(Paths.TOKEN_SERVICE_PATH)
+
+  def execute(request: HttpUriRequest): ExternalResponseData[Array[Byte]] = {
```

Comment:
what happens when an error occurs in this method?
the error is thrown?

Reply:
Oh, I found where this method is called. it's wrapped in Task.
I prefer this method to return Task. it's safer.

Reply:
Agreed. Moved it lower into the pipeline.